src/patterns/ec2-app.ts

```diff
@@ -369,6 +372,7 @@ export class GuEc2App {
     const autoScalingGroup = new GuAutoScalingGroup(scope, "AutoScalingGroup", {
       app,
       vpc,
+      instanceType: props.instanceType,
```

Comment:
Should we make this stage aware so that CODE can use a smaller instance than PROD?

Reply:
Presumably users can do this themselves without the need for direct support? Perhaps a general "stage switch" function that can be plugged in as a user desires?

Reply:
Sort of. As stage is a parameter, it's a token at compile time, so we have to use CFN Mappings. GuCDK already supports mappings (https://github.com/guardian/cdk/pull/252) 🎉 .

Reply:
In my experience some teams/apps vary instance type based on stage and others do not. As we typically run fewer instances (often a single instance) in `CODE` I don't think the cost-savings here are very significant. Consequently, I am unconvinced that we should actively encourage teams to vary instance type based on stage*, as every difference between `PROD` and `CODE` introduces more possibilities for stage-specific bugs. 

As it stands users would need to do [something like this](https://github.com/guardian/cdk-playground/blob/700edcdb5298e966ef084af226c1f5ff291bd37c/cdk/lib/cdk-playground.ts#L23-L34), so I don't think the workaround is too problematic for teams who feel strongly about this.

Reply:
Your reasoning makes sense @jacobwinch and the Prism data (e.g. `prism deploy -f 'specification.instanceType' | sort -k 3`) seems to back it up.

I wonder if it's worth documenting that work around (also happy to leave it until it's needed).

Reply:
I think it would be better to document that in its own right, rather than specifically linking it to this change - perhaps [here](https://github.com/guardian/cdk/tree/main/docs) or in a more general FAQs section? I will do that in a new PR as I've also noticed some other docs that need to be updated whilst thinking about this!